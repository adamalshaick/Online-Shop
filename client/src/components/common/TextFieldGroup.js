import React from "react";
import classnames from "classnames";
import PropTypes from "prop-types";
import TextField from "@material-ui/core/TextField";

const TextFieldGroup = ({
  name,
  placeholder,
  value,
  error,
  info,
  type,
  onChange,
  disabled
}) => {
  return (
    <div>
      <TextField
        style={{ width: "100%" }}
        label={placeholder}
        name={name}
        value={value}
        onChange={onChange}
        multiline
        rows="4"
        margin="normal"
      />
      {error ? (
        <div>
          <small style={{ color: "red" }} className="entry">
            {error}
          </small>
        </div>
      ) : null}
    </div>
  );
};

TextFieldGroup.propTypes = {
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  info: PropTypes.string,
  error: PropTypes.string,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.string
};

TextFieldGroup.defaultProps = {
  type: "text"
};

export default TextFieldGroup;
