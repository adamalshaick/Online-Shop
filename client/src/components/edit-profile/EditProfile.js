import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import TextFieldGroup from "../common/TextFieldGroup";
import TextAreaFieldGroup from "../common/TextAreaFieldGroup";
import { createProfile, getCurrentProfile } from "../../actions/profileActions";
import isEmpty from "../../validation/is-empty";
import Navbar from "../layout/Navbar";

export class EditProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      handle: "",
      location: "",
      bio: "",
      errors: {}
    };
  }

  componentDidMount() {
    this.props.getCurrentProfile();
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({ errors: nextProps.errors });
    }

    if (nextProps.profile.profile) {
      const profile = nextProps.profile.profile;

      // If profile field doesnt exist, make empty string
      profile.location = !isEmpty(profile.location) ? profile.location : "";
      profile.bio = !isEmpty(profile.bio) ? profile.bio : "";

      // Set component fields state
      this.setState({
        handle: profile.handle,
        location: profile.location,
        bio: profile.bio
      });
    }
  }

  onSubmit = e => {
    e.preventDefault();

    const profileData = {
      handle: this.state.handle,
      location: this.state.location,
      bio: this.state.bio
    };
    this.props.createProfile(profileData, this.props.history);
  };

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  render() {
    const { errors } = this.state;
    return (
      <>
        <Navbar />
        <div className="mt-5 entry container">
          <div className="row">
            <div
              style={{
                boxShadow:
                  "0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
                padding: "2rem"
              }}
              className="col-md-8 m-auto"
            >
              <h1 style={{ fontSize: "1.7rem" }} className="text-center mb-4">
                Edit Your Profile
              </h1>

              <small className="d-block pb-3">* = required fields</small>
              <form onSubmit={this.onSubmit}>
                <TextFieldGroup
                  id="handle"
                  placeholder="* Profile Handle"
                  name="handle"
                  value={this.state.handle}
                  onChange={this.onChange}
                  error={errors.handle}
                  info="Your username"
                />

                <TextFieldGroup
                  id="location"
                  placeholder="Location"
                  name="location"
                  value={this.state.location}
                  onChange={this.onChange}
                  error={errors.location}
                  info="Your Location"
                />

                <TextAreaFieldGroup
                  id="bio"
                  placeholder="Short Bio"
                  name="bio"
                  value={this.state.bio}
                  onChange={this.onChange}
                  error={errors.bio}
                  info="Tell us a little about yourself"
                />
                <input
                  type="submit"
                  value="Submit"
                  className="btn btn-dark float-right mt-4"
                />
              </form>
            </div>
          </div>
        </div>
      </>
    );
  }
}

EditProfile.propTypes = {
  profile: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
  createProfile: PropTypes.func.isRequired,
  getCurrentProfile: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  profile: state.profile,
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { createProfile, getCurrentProfile }
)(EditProfile);
