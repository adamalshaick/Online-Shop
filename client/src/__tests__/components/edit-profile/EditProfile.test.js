import React from "react";
import { mount } from "enzyme";
import { shallow } from "enzyme";
import EditProfile from "../../../components/edit-profile/EditProfile";
import store from "../../../store";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import TextFieldGroup from "../../../components/common/TextFieldGroup";

const wrapped = mount(
  <Router>
    <EditProfile store={store} />
  </Router>
);
// const component = wrapped.dive();

it("has a text area and a button", () => {
  //   expect(wrapped.find(<TextFieldGroup />).length).toEqual(0);
  //   //   expect(component.find("button").length).toEqual(2);
});

wrapped.unmount();
