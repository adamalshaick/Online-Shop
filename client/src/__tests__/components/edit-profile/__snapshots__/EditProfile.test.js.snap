// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<Fragment>
  <Connect(Navbar) />
  <div
    className="mt-5 entry container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-8 m-auto"
        style={
          Object {
            "boxShadow": "0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
            "padding": "2rem",
          }
        }
      >
        <h1
          className="text-center mb-4"
          style={
            Object {
              "fontSize": "1.7rem",
            }
          }
        >
          Edit Your Profile
        </h1>
        <small
          className="d-block pb-3"
        >
          * = required fields
        </small>
        <form
          onSubmit={[Function]}
        >
          <TextFieldGroup
            id="handle"
            info="Your username"
            name="handle"
            onChange={[Function]}
            placeholder="* Profile Handle"
            type="text"
            value="handle value"
          />
          <TextFieldGroup
            id="location"
            info="Your Location"
            name="location"
            onChange={[Function]}
            placeholder="Location"
            type="text"
            value="location value"
          />
          <TextAreaFieldGroup
            id="bio"
            info="Tell us a little about yourself"
            name="bio"
            onChange={[Function]}
            placeholder="Short Bio"
            value="bio value"
          />
          <input
            className="btn btn-dark float-right mt-4"
            type="submit"
            value="Submit"
          />
        </form>
      </div>
    </div>
  </div>
</Fragment>
`;
